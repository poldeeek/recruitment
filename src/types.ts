export interface IFetchData {
  project: string;
  course: string;
  person: string;
  email: string;
  department: string;
  location: string;
  courseStartedDate: string;
  openedLessonsCount: string;
  completedDate: string;
  completedLessonsCount: number;
  totalLessonsCount: number;
  haveNotStarted: string;
  notOnSchedule: string;
  haveStarted: string;
  quizScore: number;
  quizScoreTotal: number;
  certificateTitle: string | null;
}

export interface ICourse {
  course: string;
  openedLessonsCount: number;
  projects: Array<ProjectType>;
}

export type ProjectType = {
  project: string;
  completedLessonsCount: number;
};
