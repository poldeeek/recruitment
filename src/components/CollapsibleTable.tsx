import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

import Row from './Row';

import { ICourse } from '../types';

interface ICollapsibleTable {
  courses: Array<ICourse>;
}

const CollapsibleTable: React.FC<ICollapsibleTable> = ({ courses }) => {
  if (!courses) {
    return <div>No courses</div>;
  }

  return (
    <TableContainer component={Paper}>
      <Table aria-label="collapsible table">
        <TableHead>
          <TableRow>
            <TableCell />
            <TableCell>
              <Typography variant="h4" component="h4">
                Course Name
              </Typography>
            </TableCell>
            <TableCell align="right">
              <Typography variant="h4" component="h4">
                Opened Lessons
              </Typography>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {courses.map((course: ICourse) => (
            <Row key={course.course} course={course} />
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default CollapsibleTable;
