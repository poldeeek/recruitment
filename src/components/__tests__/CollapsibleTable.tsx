import '@testing-library/jest-dom/extend-expect';
import CollapsibleTable from '../CollapsibleTable';
import { render } from '@testing-library/react';
import { ICourse } from '../../types';

jest.mock('../Row.tsx', () => {
  const Row: React.FC<ICourse> = () => (
    <tr>
      <td>Course</td>
    </tr>
  );
  return Row;
});

const props = [
  {
    course: 'Course1',
    openedLessonsCount: 8,
    projects: [],
  },
  {
    course: 'Course2',
    openedLessonsCount: 1,
    projects: [],
  },
];

test('Basic render', () => {
  const component = render(<CollapsibleTable courses={props} />);
  expect(component.baseElement).toMatchSnapshot();
});

test('Basic render, no courses', () => {
  const component = render(<CollapsibleTable courses={[]} />);
  expect(component.baseElement).toMatchSnapshot();
});
