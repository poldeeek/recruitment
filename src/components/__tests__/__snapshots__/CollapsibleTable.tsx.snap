// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="collapsible table"
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            />
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                Course Name
              </h4>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                Opened Lessons
              </h4>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr>
            <td>
              Course
            </td>
          </tr>
          <tr>
            <td>
              Course
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`Basic render, no courses 1`] = `
<body>
  <div>
    <div>
      No courses
    </div>
  </div>
</body>
`;
